8thwall-camera.js

```diff
@@ -5,88 +5,402 @@
+
+    view: null, // cache camera
+
+    position: [0, 0, 0], // cache 8thwall cam position
+
+    rotation: [0, 0, 0, -1], // cache 8thwall cam rotation
+
+    GlTextureRenderer: null, // cache XR8.GlTextureRenderer.pipelineModule
+
+    promptForDeviceMotion: async function () {
+        return new Promise(async (resolve, reject) => {
+            
+            // Tell anyone who's interested that we want to get some user interaction
+            window.dispatchEvent(new Event("8thwall-request-user-interaction"));
+
+            // Wait until someone response that user interaction happened
+            window.addEventListener("8thwall-permissions-allowed", async() => {
+                try {
+                    const motionEvent = await DeviceMotionEvent.requestPermission();
+                    resolve(motionEvent);
+                } catch (exception) {
+                    reject(exception)
+                }
+            });
+
+            
+        })
+    },
+
+    getPermissions: async function () {
+        // iOS "feature". If we want to request the DeviceMotion permission, user has to interact with the page at first (touch it).
+        // If there was no interaction done so far, we will render a HTML overlay with would get the user to interact with the screen
+
+        if (DeviceMotionEvent && DeviceMotionEvent.requestPermission) {
+            try {
+                const result = await DeviceMotionEvent.requestPermission();
+
+                // The user must have rejected the motion event on previous page load. (safari remembers this choice).
+                if (result !== "granted") {
+                    throw new Error("MotionEvent");
+                }
+            } catch (exception) {
 
-        const vals = ['auto', 'back', 'front'];
-        this.camera = vals[this.camera];
-        if(this.camera == 'auto') {
-            this.camera = 'back';
+                // User had no interaction with the page so far
+                if (exception.name === "NotAllowedError") {
+                    const motionEvent = await this.promptForDeviceMotion();
+                    if (motionEvent !== "granted") {
+                        throw new Error("MotionEvent");
+                    }
+                } else {
+                    throw new Error("MotionEvent");
+                }
+            }
+        }
+
+        try {
+            // make sure we get the camera stream
+            const stream = await navigator.mediaDevices.getUserMedia({ video: true, audio: true })
+
+            // If we successfully acquired the camera stream - we can stop it and wait until 8thwall requests it again
+            stream.getTracks().forEach((track) => {
+                track.stop();
+            });
+
+        } catch (exception) {
+            throw new Error("Camera");
         }
+    },
+
 
-        this.onStart = this.onStart.bind(this);
+    init: function () {
+        this.view = this.object.getComponent("view");
```

Comment:
init() does not guarantee that any other components are available or initialized yet, so maybe move this line to `start()`